kratos/processes/compute_nodal_normal_divergence_process.cpp

```diff
@@ -80,7 +94,9 @@ void ComputeNodalNormalDivergenceProcess<THistorical>::Execute()
             for(std::size_t i_node=0; i_node<number_of_nodes; ++i_node) {
                 const auto& vector_field = get_vector_field(r_geometry[i_node], *mpOriginVariable);
 
-                divergence += inner_prod( row(rTls.DN_DX, i_node), vector_field );
+                for(std::size_t k=0; k<dimension; ++k) {
+                    divergence += rTls.DN_DX(i_node, k)*vector_field[k];
+                }
```

Comment:
Prior to your change this segfaulted in 2D right?

Reply:
Mismatched sizes return error only in full-debug mode (I encountered this error only with mpirun).